### PR TITLE
Remove Channel and ChannelCounterparty traits

### LINKED ISSUE #192: Cleanup exported types in channel
> The exported module brings up a related problem I believe. Parts of this module (and its intended functionality) may be unnecessary and we can remove it..

yes, we should remove the traits from the `exported.rs` files (in both connection and channel folders).

_Originally posted by @ancazamfir in https://github.com/informalsystems/ibc-rs/issues/186#issuecomment-667995580_

More specifically, the traits `Channel` and `ChannelCounterparty` can be removed:

https://github.com/informalsystems/ibc-rs/blob/6f89b2ddbd16c3b7b77476a0ef0683b5f531fe14/modules/src/ics04_channel/exported.rs#L6-L22

### PR DESCRIPTION
**Pull request by c410-f3r**
Fixes #192

### CODE REVIEW

**Comment by c410-f3r** [comment]
Probably not related:
`Codecov failed with the following error: connect ETIMEDOUT`

File: modules/src/ics04_channel/channel.rs [hunk 1]
```diff
@@ -142,20 +133,16 @@ impl Counterparty {
...
-        self.port_id.as_str().into()
+    pub fn channel_id(&self) -> String {
+        self.channel_id.as_str().into()
     }
 
-    fn channel_id(&self) -> String {
-        self.channel_id.as_str().into()
+    pub fn port_id(&self) -> String {
+        self.port_id.as_str().into()
     }
```
Comment (ancazamfir) [inline modules/src/ics04_channel/channel.rs:143]: Any reason for changing the order of these functions?

Comment (c410-f3r) [reply modules/src/ics04_channel/channel.rs]: Mostly for organization and aesthetics. Due to the fact that this implementation block had only three methods, they were arranged in ascending natural order.

**Comment by ancazamfir** [comment]
Looks good, can you solve the conflicts so we prepare for merge?

**Comment by c410-f3r** [comment]
Done

**Pull request merged**